feature/authentication/src/main/java/com/finance2up/authentication/presentation/ui/otp/OTPScreen.kt

```diff
@@ -0,0 +1,226 @@
+package com.finance2up.authentication.presentation.ui.otp
+
+import android.annotation.SuppressLint
+import android.util.Log
+import android.widget.Toast
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.focus.FocusRequester
+import androidx.compose.ui.focus.focusRequester
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.navigation.NavController
+import com.finance2up.authentication.R
+import com.finance2up.authentication.presentation.util.fontSizeDimensionResource
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+
+@SuppressLint("UnrememberedMutableState")
+@Composable
+fun OTPScreen(navController: NavController) {
+    val otpViewModel: OTPViewModel = hiltViewModel()
+
+    val otpUIState = otpViewModel.otpUIState.collectAsStateWithLifecycle()
+    val preotpUIState = otpViewModel.preotpUIState.collectAsStateWithLifecycle()
+    val otpState = otpViewModel.otpState.collectAsState()
```

Comment:
sao mình cần có 3 UiState khác nhau cho 1 Screen nhỉ?

Reply:
vì màn nhập OTP của e cần email từ màn trước. Nên cần preotpUIState để lấy giá trị email. 

Còn otpUIState là trạng thái các giá trị lấy từ View

otpState là trạng thái của model Response từ BE ạ

Reply:
liệu trong ViewModel mình tính toán và gộp 3 cái làm 1 đc ko nhỉ? Hay bắt buộc phải để 3 cái? Thường 1 màn hình chỉ có 1 UiState thôi